docs/internal/component_dependency.txt

```diff
@@ -0,0 +1,42 @@
+
+
+
+1. class Regex: 
+- defined in regex.hpp
+- A simple regex-tree class ({EPSILON, ATOMIC, UNION, CONCAT, KPLUS, KSTAR})
+- Depends on : Nothing
+
+
+
```

Comment:
I feel, in-code documentation is better in general. That helps to keep the code and the documentation in sync. Otherwise, we might end up adding more classes that may not be reflected in the documentation here. What are your thoughts on it ?